### ISSUE TITLE
Attempt to integrate with ORCID's OIDC

### ISSUE THREAD

**Issue by brecke** [pow-auth/pow_assent#237]

**Comment by danschultzer** [comment]
There should be an error log entry strating with `Strategy failed with error:`, do you see anything?

If there's no error log then I think it can show up if registration has been disabled or there are missing params. In the latter case it would be helpful to have the logger print a warning, I'll open a PR to fix this.

**Comment by danschultzer** [comment]
If you update your deps with `{:pow_assent, git: "https://github.com/pow-auth/pow_assent.git", ref: "log-user-create-failed"}` you will maybe see an error being logged.

**Comment by brecke** [comment]
thanks @danschultzer it logs something now:

```
[notice] TLS :client: In state :wait_cert_cr at ssl_handshake.erl:2113 generated CLIENT ALERT: Fatal - Handshake Failure
 - {:bad_cert, :unable_to_match_altnames}
```

Not sure what's wrong but I'll try and look it up

**Comment by danschultzer** [comment]
That's odd, you should definitely see an `Strategy failed` error logged somewhere with an  `{:error, :failed_to_connect` tuple? Are you sure nothing is in the logs other than the TLS client notice?

This error means that the cert is invalid. What URL are you using for ORCID OIDC (this `:site` config)?

**Comment by danschultzer** [comment]
You should change the strategy base to OIDC:

```elixir
defmodule Paperlens.Orcid.OrcidProvider do
  use Assent.Strategy.OIDC.Base
```

I think the TLS warning might have been a separate thing, not related to ORCID. I haven't been able to reproduce it accessing `orcid.org`. Digging into why the error log doesn't show up, it's confusing.

**Comment by danschultzer** [comment]
Found the issue! The disabled registration flag was on, because there is no router helpers enabled (with Phoenix 1.7 it's disabled and will likely be deprecated). The router helpers was used to detect whether registration is disabled. This has been resolved in #239.

If you want to test it right away, you can use `{:pow_assent, git: "https://github.com/pow-auth/pow_assent.git", ref: "fix-verified-routes-handling"}`. I'm going to check Pow as well to make sure I'm testing everything with the router helpers disabled. Release will be out later today.

**Comment by danschultzer** [comment]
v0.4.17 released with this fix, thanks!

FWIW the provider can be made super minimal:

```elixir
defmodule Paperlens.Orcid.OrcidProvider do
  use Assent.Strategy.OIDC.Base

  @impl true
  def default_config(_config) do
    [
      site: "https://orcid.org",
      client_authentication_method: "client_secret_post"
    ]
  end
end
```

```elixir
config :paperlens, :pow_assent,
  http_adapter: Assent.HTTPAdapter.Mint,
  providers: [
    orcid: [
      client_id: System.get_env("ORCID_CLIENT_ID"),
      client_secret: System.get_env("ORCID_CLIENT_SECRET"),
      strategy: Paperlens.Orcid.OrcidProvider
    ]
  ]
```

**Comment by brecke** [comment]
Glad I helped in some way :) I'll just suggest something here: listing phoenix supported versions on the readme file so one can immediately know whether something works 100% or still under testing. Anyway, good job!

**Comment by danschultzer** [comment]
Yeah, it was supposed to work with 1.7. The deps version requirement on hex.pm shows which Pow/PowAssent version works with which Phoenix version. I just hadn’t updated the tests to the new helpers structure so everything seemed to work when I added 1.7 support 😬
